drivers/input/CMakeLists.txt

```diff
@@ -7,5 +7,12 @@ zephyr_library_sources_ifdef(CONFIG_INPUT_FT5336 input_ft5336.c)
 zephyr_library_sources_ifdef(CONFIG_INPUT_GPIO_KEYS input_gpio_keys.c)
 zephyr_library_sources_ifdef(CONFIG_INPUT_GPIO_QDEC input_gpio_qdec.c)
 zephyr_library_sources_ifdef(CONFIG_INPUT_NPCX_KBD input_npcx_kbd.c)
-zephyr_library_sources_ifdef(CONFIG_INPUT_SDL_TOUCH input_sdl_touch.c)
+if (CONFIG_INPUT_SDL_TOUCH)
+	zephyr_library_sources(input_sdl_touch.c)
+	if (CONFIG_NATIVE_APPLICATION)
+		zephyr_library_sources(input_sdl_touch_bottom.c)
+	else()
+		target_sources(native_simulator INTERFACE input_sdl_touch_bottom.c)
+	endif()
+endif()
```

Comment:
Could you move this after the block of "trivial" `zephyr_library_sources_ifdef`? I think it makes it easier to maintain the list since this is its own block of logic. I'd suggest doing it on the other subsystems as well. (I'm working on a CI check to keep these lists sorted :-))